posthog/plugins/utils.py

```diff
@@ -249,6 +249,46 @@ def get_file_from_archive(archive: bytes, filename: str, *, json_parse: bool = T
         return None
 
 
+def find_index_ts_in_archive(archive: bytes, main_filename: Optional[str] = None) -> str:
+    main_filenames_to_try = [main_filename] if main_filename else ["index.js", "index.ts"]
+    for main_filename in main_filenames_to_try:
+        if index_ts := get_file_from_archive(archive, main_filename, json_parse=False):
+            return index_ts
+    raise ValueError(f"Could not find main file {' or '.join(main_filenames_to_try)}")
+
+
+def extract_plugin_code(
+    archive: bytes, plugin_json_parsed: Optional[Dict[str, Any]] = None
+) -> Tuple[str, Optional[str], Optional[str]]:
+    """Extract plugin.json, index.ts (which can be aliased) and frontend.tsx out of an archive.
+
+        If plugin.json has already been parsed before this is called, its value can be passed in as an optimization."""
+    if archive is None:
+        raise ValueError(f"There is no archive to extract code from")
+    # Extract plugin.json - required, might be provided already
+    plugin_json: str
+    if not plugin_json_parsed:
```

Comment:
So there are two cases here: None and empty dictionary. are both of these supposed to behave identically?

Reply:
Yes, at least `name` needs to be in the object

Reply:
However, in this case it's probably better to only care about the `None` case and warn about `{}` plugin-server-side, so good catch!